pkg/rtc/room.go

```diff
@@ -985,7 +987,10 @@ func (r *Room) AddAgentDispatch(dispatch *livekit.AgentDispatch) (*livekit.Agent
 	r.lock.RLock()
 	// launchPublisherAgents starts a goroutine to send requests, so is safe to call locked
 	for _, p := range r.participants {
-		r.launchPublisherAgents([]*agentDispatch{ad}, p)
+		if p.IsPublisher() {
+			r.launchTargetAgents([]*agentDispatch{ad}, p, livekit.JobType_JT_PUBLISHER)
+		}
+		r.launchTargetAgents([]*agentDispatch{ad}, p, livekit.JobType_JT_PARTICIPANT)
```

Comment:
Asked a question earlier. Looks like publisher gets both types launches? Could there be any confusion because of two jobs for same participant?

Reply:
maybe. workers only register to handle one type of job so it doesn't seem that easy to get wrong...